http/1.7/modules/ROOT/pages/http-authentication.adoc

```diff
@@ -699,6 +700,223 @@ In the *Configuration XML* editor, the `tls:context`, `tls:trust-store` and `tls
+
+When you create a Mule app, the flow tries to access a protected resource in the Anypoint organization server with no authentication configured:
+
+[source,xml,linenums]
+----
+<http:listener-config name="listenerConfig">
+  <http:listener-connection host="0.0.0.0" port="8070" />
+</http:listener-config>
+
+<http:request-config name="requestConfig">
+  <http:request-connection protocol="HTTPS" host="${host}" port="${port}" />
+</http:request-config>
+
+<flow name="authCodeFlow">
+  <http:listener config-ref="listenerConfig" path="/test" />
+  <http:request method="GET" config-ref="requestConfig" path="${resource}" />
+</flow>
+----
+
+The Mule app consists of an HTTP *Listener* source and an HTTP *Request* operation. If the `host`, `port`, and `resource` properties in your YAML file aren't set to an accessible resource, the Anypoint platform fails with a 401 status code because it expects to obtain the access token in the request:
+
+[source,YAML,linenums]
+----
+host: "anypoint.mulesoft.com"
+port: "443"
+resource: "/apiplatform/repository/v2/organizations/<your-org-id>/environments"
+----
+
+=== Add Redirection to Local Authorization URL
+
+The app uses credentials to identify itself to the authentication server. On your XML file, add the redirection to the authorization URL:
+
+[source,xml,linenums]
+----
+<http:listener-config name="listenerConfig">
+  <http:listener-connection host="0.0.0.0" port="8070" />
+</http:listener-config>
+
+<http:request-config name="requestConfig">
+  <http:request-connection protocol="HTTPS" host="${host}" port="${port}" />
+</http:request-config>
+
+<flow name="authCodeFlow">
+  <http:listener config-ref="listenerConfig" path="/test" >
+    <http:response statusCode="#[vars.redirectStatus default '200']" >
+      <http:headers>
+        #[vars.redirectLocationHeaders default {}]
+      </http:headers>
+    </http:response>
+  </http:listener>
+ 
+ <http:request method="GET" config-ref="requestConfig" path="${resource}" />
+  <error-handler>
+    <on-error-continue type="MULE:SECURITY" >
+      <set-variable
+        variableName="redirectLocationHeaders" 
+        value="#[{'Location': '${local.authorization.url}'}]" />
+      <set-variable variableName="redirectStatus" value="302" />
+    </on-error-continue>
+  </error-handler>
+</flow>
+----
+
+Do the same to your YAML file:
+
+[source,YAML,linenums]
+----
+host: "anypoint.mulesoft.com"
+port: "443"
+resource: "/apiplatform/repository/v2/organizations/<your-org-id>/environments"
+local.authorization.url: "http://localhost:8070/authorize"
+----
+
+=== Add OAuth Authentication
+
+An OAuth dance mechanism starts when the authentication is added to the HTTP request connection. This creates an endpoint in the `localAuthorizationUrl` that redirects you to the `authorizationUrl` parameter used to configure the authorization server location. In the following example, the location is `https://anypoint.mulesoft.com/accounts/api/v2/oauth2/authorize`.
```

Comment:
Maybe in this point we could clarify the detail regarding the dynamic configs. What do you think?

Reply:
What do you think of:
"Note that this example uses dynamic configuration. See xref:mule-sdk::static-dynamic-configs.adoc[Dynamic Configurations]."

Reply:
It's ok for me!